src/ur5e_arm.cpp

```diff
@@ -412,5 +412,10 @@ void UR5eArm::read_and_noop() {
 
         // send a noop to keep the connection alive
         driver->writeTrajectoryControlMessage(control::TrajectoryControlMessage::TRAJECTORY_NOOP, 0, RobotReceiveTimeout::off());
+    } else {
```

Comment:
hmm it seems a little funny to be doing this resetting of communications in a function called `read_and_noop`. Thats definitely not something I would expect buried here if I were new to the codebase. Maybe its time for a better name for the function to communicate this?  The alternative to that would be erroring here and catching with this in the calling code.  That would be more verbose and I'm not sure more useful though.

Reply:
thats fair, the concern I had with returning the error/handling this somewhere else was because read_and_noop gets used by multiple threads. would feel better about that if only keep alive was calling into there. 

maybe we could just rename this to read_from_driver? ngl I had too much fun calling the function noop that I don't remember what noop stood for. 

@nicksanford since you are also messing with these functions, not sure if you had something in mind already.

Reply:
I like it!